io/plugins/eu.esdihumboldt.hale.io.csv/src/eu/esdihumboldt/hale/io/csv/writer/internal/CSVInstanceWriter.java

```diff
@@ -138,12 +176,10 @@ protected IOReport execute(ProgressIndicator progress, IOReporter reporter)
 
 			// header is only finished if all properties have been processed
 			// insert header to temp file and write it to output
-			insertHeader(tempFile, getTarget().getOutput(), headerRow);
+			insertHeader(tempFile, new FileOutputStream(filenameWithType), headerRow);
```

Comment:
This breaks compatibility with the previous functionality, because there it was possible to use something as a target that is not a file.
It would be great if that behavior could be kept at least for the case where only one type is selected and thus your `changeName` variable set to `false`.

Reply:
Would you mind telling me how to have a target which is not a file to test?
Thank you

Reply:
If you want to test this in a unit test you could use a `ByteArrayOutputStream` as target.

Reply:
I would guess that what you meant is that at the [line](https://github.com/halestudio/hale/blob/ce740b79fa0f04d371abe6f9c61776f62534c0c5/io/plugins/eu.esdihumboldt.hale.io.csv.test/src/eu/esdihumboldt/hale/io/csv/writer/CSVInstanceWriterTest.java#L331C12-L331C12) could be send a supplier of a file `FileIOSupplier `or a `ByteArrayOutputStream`. 
Any tips on how to convert a `ByteArrayOutputStream` into a `FileIOSupplier `? 
Thank you

Reply:
It shouldn't be a `FileIOSupplier`. I assume it is about [setting the writer target](https://github.com/halestudio/hale/blob/ce740b79fa0f04d371abe6f9c61776f62534c0c5/common/plugins/eu.esdihumboldt.hale.common.core/src/eu/esdihumboldt/hale/common/core/io/ExportProvider.java#L42)?
There the type is `LocatableOutputSupplier`. You could check what existing implementations are there that would fit and if there aren't any create one by implementing the interface.